utils/wandb_logging/wandb_utils.py

```diff
@@ -0,0 +1,172 @@
+    print(f"{colorstr('wandb: ')}Install Weights & Biases for YOLOv5 logging with 'pip install wandb' (recommended)")
+
+WANDB_ARTIFACT_PREFIX = 'wandb-artifact://'
+
+def remove_prefix(from_string, prefix):
+    return from_string[len(prefix):]
+    
+class WandbLogger():
+    def __init__(self, opt, name, run_id, data_dict, job_type='Training'):
+        self.wandb = wandb
+        if self.wandb:
+            self.wandb_run = wandb.init(config=opt, resume="allow",
+                                   project='YOLOv5' if opt.project == 'runs/train' else Path(opt.project).stem,
+                                   name=name,
+                                   job_type=job_type,
+                                   id=run_id)
+        else:
+            self.wandb_run = None
+        if job_type == 'Training':
+            self.setup_training(opt, data_dict)
+            if opt.bbox_interval == -1:
+                opt.bbox_interval = (opt.epochs // 10) if opt.epochs > 10 else opt.epochs
+            if opt.save_period == -1:
+                opt.save_period = (opt.epochs // 10) if opt.epochs > 10 else opt.epochs
+    
+    def setup_training(self, opt, data_dict):
+        self.log_dict = {}
+        self.train_artifact_path, self.trainset_artifact = self.download_dataset_artifact(data_dict['train'], opt.artifact_alias)
+        self.test_artifact_path, self.testset_artifact = self.download_dataset_artifact(data_dict['val'], opt.artifact_alias)
+        self.result_artifact, self.result_table, self.weights = None, None, None
+        if self.train_artifact_path is not None:
+            train_path = self.train_artifact_path + '/data/images/'
+            data_dict['train'] = train_path
+        if self.test_artifact_path is not None:
+            test_path = self.test_artifact_path + '/data/images/'
+            data_dict['val'] = test_path
+            self.result_artifact = wandb.Artifact("run_"+wandb.run.id+"_progress", "evaluation")
+            self.result_table = wandb.Table(["epoch", "id", "prediction", "avg_confidence"])
+        if opt.resume_from_artifact:
+            modeldir, _ = self.download_model_artifact(opt.resume_from_artifact)
+            if modeldir:
+                self.weights = modeldir + "/best.pt"
+                opt.weights = self.weights
+        
+
+    def download_dataset_artifact(self,path, alias):
+        if path.startswith(WANDB_ARTIFACT_PREFIX):
+            dataset_artifact = wandb.use_artifact(remove_prefix(path,WANDB_ARTIFACT_PREFIX)+":"+alias)
+            if dataset_artifact is None:
+                logger.error('Error: W&B dataset artifact doesn\'t exist')
+                raise ValueError('Artifact doesn\'t exist')
+            datadir = dataset_artifact.download()
+            labels_zip = datadir+"/data/labels.zip"
+            shutil.unpack_archive(labels_zip, datadir+'/data/labels', 'zip')
+            print("Downloaded dataset to : ", datadir)
+            return datadir, dataset_artifact
```

Comment:
Here the images and labels are downloaded as artifacts. And these files will then be loaded by train.py itself in traditional fashion, using correct stride.